glean-core/src/debug.rs

```diff
@@ -128,17 +157,30 @@ where
     }
 }
 
-/// The debug view tag is the value for the `X-Debug-ID` header of tagged ping requests,
-/// thus is it must be a valid header value.
+fn get_bool_from_str(value: String) -> Option<bool> {
+    std::str::FromStr::from_str(&value).ok()
+}
+
+fn tokenize_string(value: String) -> Option<Vec<String>> {
+    let trimmed = value.trim();
+    if trimmed.is_empty() {
+        return None;
+    }
+
+    Some(trimmed.split(',').map(|s| s.trim().to_string()).collect())
+}
+
+/// A tag is the value used in both the `X-Debug-ID` and `X-Source-Tags` headers
+/// of tagged ping requests, thus is it must be a valid header value.
 ///
 /// In other words, it must match the regex: "[a-zA-Z0-9-]{1,20}"
 ///
 /// The regex crate isn't used here because it adds to the binary size,
 /// and the Glean SDK doesn't use regular expressions anywhere else.
-pub fn validate_debug_view_tag(value: String) -> Option<String> {
+fn validate_tag(value: &String) -> bool {
```

Comment:
To not make Clippy unhappy here (we need to use `&String` if we don't want to do more generic shenanigans)

```suggestion
#[allow(clippy::ptr_arg)]
fn validate_tag(value: &String) -> bool {
```